source/GameData.cpp

```diff
@@ -161,14 +161,14 @@ namespace {
 
 		// Try adding all the possible icon variants.
 		if(Files::Exists(path + "icon.png"))
-			icon->Add(path + "icon.png");
+			icon->Add(filesystem::path(path + "icon.png"));
 		else if(Files::Exists(path + "icon.jpg"))
-			icon->Add(path + "icon.jpg");
+			icon->Add(filesystem::path(path + "icon.jpg"));
 
 		if(Files::Exists(path + "icon@2x.png"))
-			icon->Add(path + "icon@2x.png");
+			icon->Add(filesystem::path(path + "icon@2x.png"));
 		else if(Files::Exists(path + "icon@2x.jpg"))
-			icon->Add(path + "icon@2x.jpg");
+			icon->Add(filesystem::path(path + "icon@2x.jpg"));
```

Comment:
Probably not great to support `.jpeg` and `.jpe` elsewhere but not here.
Maybe something like:
```c++
for(const string &extension : ImageBuffer::ImageExtensions())
{
    string iconPath = path + "icon." + extension;
    if(File::Exists(iconPath))
    {
        // whatever
    }
}
```